src/scion/interpolation/Histogram/test/Histogram.test.cpp

```diff
@@ -31,6 +31,29 @@ SCENARIO( "Histogram" ) {
         CHECK_THAT( 1., WithinRel( interpolator( 2.0, xLeft, xRight, yLeft, yRight ) ) );
       } // THEN
     } // WHEN
+
+    WHEN( "interpolating an x,f(y) interval" ) {
+
+      interpolation::Histogram interpolator{};
+
+      THEN( "the interpolation is performed correctly" ) {
+
+        double xLeft = 1.0;
+        double xRight = 2.0;
+        auto fLeft = [] ( double y ) { return y; };
+        auto fRight = [] ( double y ) { return - y; };
+
+        CHECK_THAT(  0.0, WithinRel( interpolator( 1.0,  0.0, xLeft, xRight, fLeft, fRight ) ) );
+        CHECK_THAT(  0.0, WithinRel( interpolator( 1.5,  0.0, xLeft, xRight, fLeft, fRight ) ) );
+        CHECK_THAT(  0.0, WithinRel( interpolator( 2.0,  0.0, xLeft, xRight, fLeft, fRight ) ) );
+        CHECK_THAT(  1.0, WithinRel( interpolator( 1.0,  1.0, xLeft, xRight, fLeft, fRight ) ) );
+        CHECK_THAT(  1.0, WithinRel( interpolator( 1.5,  1.0, xLeft, xRight, fLeft, fRight ) ) );
+        CHECK_THAT(  1.0, WithinRel( interpolator( 2.0,  1.0, xLeft, xRight, fLeft, fRight ) ) );
+        CHECK_THAT( -1.0, WithinRel( interpolator( 1.0, -1.0, xLeft, xRight, fLeft, fRight ) ) );
+        CHECK_THAT( -1.0, WithinRel( interpolator( 1.5, -1.0, xLeft, xRight, fLeft, fRight ) ) );
+        CHECK_THAT( -1.0, WithinRel( interpolator( 2.0, -1.0, xLeft, xRight, fLeft, fRight ) ) );
+      } // THEN
+    } // WHEN
```

Comment:
I can see that the values being returned are the ones expected... what I don't get is the logic behind the return. in Histogram.hpp the interpolation is just returning an unaltered yLeft. It is not modified by or influenced by any of the other inputs as far as I can see. In this case, whatever fLeft returns will be the return of the interpolation and it's not clear to me that this is an interpolation

Reply:
These interpolation functors are used to interpolate data on intervals defined by a grid. In case of the 1D version, you'll have 2 grid points (xLeft and yLeft) and 2 function values (yLeft and yRight). For the 2D version you'll have 2 grid points (xLeft and yLeft) and 2 actual functions (fLeft and fRight, which describe the function z = f(xLeft, y) and z = f(xRight, y)).

When we say we are using "histogram" or "constant" interpolation over such an interval, we mean that the value is constant over the interval, i.e. the left side value is returned. This interpolation type is rarely used in evaluated nuclear data these days - and I prefer using a linerised step-wise function to represent constant interpolation.